fs/shm/Kconfig

```diff
@@ -3,14 +3,15 @@
 # see the file kconfig-language.txt in the NuttX tools repository.
 #
 
-config FS_SHM
+config FS_SHMFS
 	bool "Shared memory support"
+	select MM_MAP
+	select MM_SHM if BUILD_KERNEL
```

Comment:
I am still not quite happy with this one. Should we instead have common virtual page allocator always present in CONFIG_BUILD_KERNEL, and replace shm_alloc/shm_free with that? That is, move those functions out of mm/shm (into mm/map?) so that you don't need to pull in the system-v shm everywhere where you need to allocate vm for pages....

Reply:
Fine by me of course

Reply:
how would the few first items in this PR look to your eye regarding the common virtual page allocator?